### PR TITLE
Refactor/Rework Admin Item Functionality

### PR DESCRIPTION
**Pull request by steve-meyers**
- Fix routes
- Add AdminItemsController
- Rework spec files to follow updated routes
- Remove unnecessary logic from ItemsController (leftover from brown code)
- Fix Admin Merchant Show page
- Fix Admin Merchant Item index 
- Delete duplicate tests from brown code
- [x] All tests passing?

### CODE REVIEW

File: app/controllers/admin/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Admin::ItemsController < Admin::BaseController
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
```
Comment (zachholcomb) [inline app/controllers/admin/items_controller.rb:3]: we didn't have a controller that was for admin/merchants/:id/items, so we added a controller. this allows for admin to see a different page, originally the merchant items was redirecting back to the items index

File: app/views/admin/items/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<h1><%= @merchant.name%> Items</h1>
```
Comment (zachholcomb) [inline app/views/admin/items/index.html.erb:1]: this should probably be the admin/merchant/merchant_id/items/index.html.erb, but we couldn't figure out the nested resource

File: app/views/admin/merchants/show.html.erb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <p class="address"><%= @merchant.address %></p>
 <p class="address"><%= @merchant.city %>, <%= @merchant.state %> <%= @merchant.zip %></p>
 
-<ul><%= link_to "All #{@merchant.name} Items", "/merchants/#{@merchant.id}/items" %></ul>
+<ul><%= link_to "All #{@merchant.name} Items", "/admin/merchants/#{@merchant.id}/items" %></ul>
```
Comment (zachholcomb) [inline app/views/admin/merchants/show.html.erb:5]: this link was missing /admin in front

File: app/views/items/index.html.erb [hunk 4]
```diff
@@ -1,25 +1,13 @@
...
-  <% @items.each do |item| %>
+  <% @items.active_items.each do |item| %>
     <section class = "grid-item" id= 'item-<%=item.id%>'>
       <h2><%=link_to item.name, "/items/#{item.id}" %></h2>
       <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
       <%=link_to image_tag(item.image), "/items/#{item.id}" %>
       <p> <%= item.description unless @merchant%></p>
       <p>Price: <%=number_to_currency(item.price) %></p>
       <p>Inventory: <%= item.inventory %></p>
-      <% if !@merchant %>
```
Comment (zachholcomb) [inline app/views/items/index.html.erb:16]: didn't need this in the regular items index page

File: spec/features/admin/merchants/index_spec.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@
     end 
     
     it "then I see a 'disable' button next to active mechants that, when clicked, returns me to admin merchant index where I see merchant account disabled, flash message, and '/merchants/:id/items show inactive." do 
-      visit admin_merchants_path 
+      visit "/admin/merchants"
```
Comment (zachholcomb) [inline spec/features/admin/merchants/index_spec.rb:22]: this is more clear for testing

File: app/controllers/admin/items_controller.rb (hunk 1 above)
Comment (elguapogordo) [reply app/controllers/admin/items_controller.rb:3]: i like it

**Review by elguapogordo — APPROVED** [review]

File: app/controllers/items_controller.rb [hunk 6]
```diff
@@ -1,60 +1,11 @@
...
-      @items = Item.active_items
-    end
+    @items = Item.all
   end
 
   def show
     @item = Item.find(params[:id])
   end
 
-  def new
```
Comment (zachholcomb) [inline app/controllers/items_controller.rb:16]: all these were duplicate actions, merchants and admins are the only users who can create, edit and destroy items so there was no need for all these extra actions to be here

File: app/views/items/show.html.erb [hunk 7]
```diff
@@ -18,9 +18,9 @@
         <p style= "color:red">Inactive</p>
       <% end %>
       <%= link_to "Add Review", "/items/#{@item.id}/reviews/new" %>
-      <% if current_merchant? || current_admin? %>
-        <p><%= link_to "Delete Item", "/items/#{@item.id}", method: :delete if @item.no_orders?%></p>
-        <p><%= link_to "Edit Item", "/items/#{@item.id}/edit" %></p>
+      <% if (current_merchant? || current_admin?) && @item.merchant_id == current_user.id %>
```
Comment (zachholcomb) [inline app/views/items/show.html.erb:21]: this addresses the bug of merchants being able to edit/delete other merchants items

**Pull request merged**